From 466b17bb30113f7a88f17cc98e4a8cc46067a130 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Mon, 15 May 2017 15:53:37 +0300
Subject: [PATCH 1018/1345] fix: crypto: inside-secure: use Crypto API enqueue
 non generic routines

commit  f8c3bdac2c4ae3125b10bdef4d0176d5f6e28dd2 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

both ablkcipher and ahash have specific enqueue_request flavors, for
non generic code they should be used

there is no functional impact due to this change, the generic
routines invoke the non-generic routines without any impact.

Change-Id: Icc924e13b98fcec1e1cfcb6cee8847272ac398f8
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39482
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39536
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/inside-secure/cipher.c |    6 +++---
 drivers/crypto/inside-secure/hash.c   |    6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/crypto/inside-secure/cipher.c b/drivers/crypto/inside-secure/cipher.c
index 0111314..fa7534c 100644
--- a/drivers/crypto/inside-secure/cipher.c
+++ b/drivers/crypto/inside-secure/cipher.c
@@ -341,7 +341,7 @@ static int safexcel_handle_inv_result(struct safexcel_crypto_priv *priv,
 	ctx->base.send = safexcel_aes_send;
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	enq_ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, async);
+	enq_ret = ablkcipher_enqueue_request(&priv->ring[ctx->base.ring].queue, req);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 	if (enq_ret != -EINPROGRESS)
@@ -400,7 +400,7 @@ static int safexcel_cipher_exit_inv(struct crypto_tfm *tfm)
 	ctx->base.send = safexcel_cipher_send_inv;
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, &req.base);
+	ret = ablkcipher_enqueue_request(&priv->ring[ctx->base.ring].queue, &req);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 	queue_work(priv->ring[ctx->base.ring].workqueue,
@@ -452,7 +452,7 @@ static int safexcel_aes(struct ablkcipher_request *req,
 	}
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, &req->base);
+	ret = ablkcipher_enqueue_request(&priv->ring[ctx->base.ring].queue, req);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 	queue_work(priv->ring[ctx->base.ring].workqueue,
diff --git a/drivers/crypto/inside-secure/hash.c b/drivers/crypto/inside-secure/hash.c
index 3458702..0574d52 100644
--- a/drivers/crypto/inside-secure/hash.c
+++ b/drivers/crypto/inside-secure/hash.c
@@ -369,7 +369,7 @@ static int safexcel_handle_inv_result(struct safexcel_crypto_priv *priv,
 	ctx->base.send = safexcel_ahash_send;
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	enq_ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, async);
+	enq_ret = ahash_enqueue_request(&priv->ring[ctx->base.ring].queue, areq);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 
@@ -427,7 +427,7 @@ static int safexcel_ahash_exit_inv(struct crypto_tfm *tfm)
 	ctx->base.send = safexcel_ahash_send_inv;
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, &req.base);
+	ret = ahash_enqueue_request(&priv->ring[ctx->base.ring].queue, &req);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 	queue_work(priv->ring[ctx->base.ring].workqueue,
@@ -556,7 +556,7 @@ static int safexcel_ahash_update(struct ahash_request *areq)
 	}
 
 	spin_lock_bh(&priv->ring[ctx->base.ring].queue_lock);
-	ret = crypto_enqueue_request(&priv->ring[ctx->base.ring].queue, &areq->base);
+	ret = ahash_enqueue_request(&priv->ring[ctx->base.ring].queue, areq);
 	spin_unlock_bh(&priv->ring[ctx->base.ring].queue_lock);
 
 	queue_work(priv->ring[ctx->base.ring].workqueue,
-- 
1.7.9.5

